src/backend/cdb/cdbpullup.c

```diff
@@ -382,17 +382,14 @@ cdbpullup_findEclassInTargetList(EquivalenceClass *eclass, List *targetlist,
 			}
 			else
 			{
-				/* ignore RelabelType nodes on both sides */
-				while (key && IsA(key, RelabelType))
-					key = (Expr *) ((RelabelType *) key)->arg;
-
-				if (equal(tlexpr, key))
+				if ((tlexpr || key) && equal(tlexpr, key))
```

Comment:
Why did you add additional conditions?

https://github.com/arenadata/gpdb/blob/3b5aa3e9d5141832d94172e20e41a75770c07f34/src/backend/nodes/equalfuncs.c#L2843-L2849

Reply:
So that the behavior is not changed. Without this check, if `key = NULL` and `tlexpr = NULL`, `NULL` will be returned without processing remaining values from targetlist.

Reply:
if 

> behavior is not changed



> Why did you add additional conditions?

Reply:
Is this a refactoring, not a bug fix?